File: javase/src/cn/gl/day07/object/ObjectDemo3.java
package cn.gl.day07.object;

public class ObjectDemo3 {
    //属性
    int k;//变量---成员变量

    //方法
    public void m(){}
    public static void main(String[] args) {
        int k=1;//变量---局部变量
        //定义变量
        //在哪（{}）定义变量在哪使用
        /*int x=1;
        int y=2;*/
        //局部（方法）代码块---改变了变量的生命周期
        {
            int x=1;//局部变量
            int y=2;
            System.out.println(x+y);
        }
        //1000行代码与前面的x，y都没有关系
    }
}

File: J_scala/src/cn/gl/Demo/objectx/ObjectDemo3.scala
package cn.gl.Demo.objectx

object ObjectDemo3 {

  def main(args: Array[String]): Unit = {

    var t = new Teacher("Amy", 29, "female")

  }

}

// 需要这个类只能对外提供含参构造，有需要这个类中的属性可以改
// 将无参主构造私有
class Teacher private() {

  private var name: String = _
  private var age: Int = _
  private var gender: String = _

  // 提供含参辅构造
  def this(name: String, age: Int, gender: String) {
    this()
    this.name = name
    this.age = age
    this.gender = gender
  }

  def setName(name: String) = this.name = name

}
